src/app/storage/components/AddEditStorageModal/AddEditStorageForm.tsx

```diff
@@ -1,18 +1,24 @@
 import React, { useState, useContext } from 'react';
 import { Grid, GridItem } from '@patternfly/react-core';
 import Select from 'react-select';
-import AWSForm from './ProviderForms/AWSForm';
+import S3Form from './ProviderForms/S3Form';
 import GCPForm from './ProviderForms/GCPForm';
 import AzureForm from './ProviderForms/AzureForm';
 import { StorageContext } from '../../../home/duck/context';
+import { IStorage } from '../../../../models';
 const styles = require('./AddEditStorageForm.module');
 
 interface IOtherProps {
   onAddEditSubmit: any;
   onClose: any;
   addEditStatus: any;
   checkConnection: (name) => void;
-  storageList: any;
+  storageList: IStorage[];
+}
+
+interface IProviderOption {
+  label: string;
+  value: 'aws-s3' | 'generic-s3' | 'gcp' | 'azure';
```

Comment:
Interesting, didn't know this was possible.

Reply:
Yep, union types. I'm using them here instead of an enum just for brevity, but they can also be used for polymorphism since you can union values of multiple types

Reply:
Interesting stuff I was reading on this: https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant